metronome/networking/src/io/iohk/metronome/networking/RemoteConnectionManager.scala

```diff
@@ -0,0 +1,353 @@
+  def sendMessage(
+      recipient: K,
+      message: M
+  ): F[Either[ConnectionHandler.ConnectionAlreadyClosedException[K], Unit]] = {
+    connectionHandler.sendMessage(recipient, message)
+  }
+}
+//TODO add logging
+object RemoteConnectionManager {
+  case class ConnectionSuccess[F[_], K, M](
+      encryptedConnection: EncryptedConnection[F, K, M]
+  )
+
+  case class ConnectionFailure[K](
+      connectionRequest: OutGoingConnectionRequest[K],
+      err: Throwable
+  )
+
+  private def connectTo[
+      F[_]: Sync,
+      K: Codec,
+      M: Codec
+  ](
+      encryptedConnectionProvider: EncryptedConnectionProvider[F, K, M],
+      connectionRequest: OutGoingConnectionRequest[K]
+  ): F[Either[ConnectionFailure[K], ConnectionSuccess[F, K, M]]] = {
+    encryptedConnectionProvider
+      .connectTo(connectionRequest.key, connectionRequest.address)
+      .redeemWith(
+        e => Sync[F].pure(Left(ConnectionFailure(connectionRequest, e))),
+        connection => Sync[F].pure(Right(ConnectionSuccess(connection)))
+      )
+  }
+
+  case class RetryConfig(
+      initialDelay: FiniteDuration,
+      backOffFactor: Long,
+      maxDelay: FiniteDuration,
+      randomJitterConfig: RandomJitterConfig
+  )
+
+  object RetryConfig {
+    sealed abstract case class RandomJitterConfig private (
+        fractionOfDelay: Double
+    )
+
+    object RandomJitterConfig {
+      import scala.concurrent.duration._
+
+      /** Build random jitter config
+        * @param fractionOfTheDelay, in what range in the computed jitter should lay, it should by in range 0..1
+        */
+      def buildJitterConfig(
+          fractionOfTheDelay: Double
+      ): Option[RandomJitterConfig] = {
+        if (fractionOfTheDelay >= 0 && fractionOfTheDelay <= 1) {
+          Some(new RandomJitterConfig(fractionOfTheDelay) {})
+        } else {
+          None
+        }
+      }
+
+      /** computes new duration with additional random jitter added. Works with millisecond precision i.e if provided duration
+        * will be less than 1 millisecond then no jitter will be added
+        * @param config,jitter config
+        * @param delay, duration to randomize it should positive number otherwise no randomization will happen
+        */
+      def randomizeWithJitter(
+          config: RandomJitterConfig,
+          delay: FiniteDuration
+      ): FiniteDuration = {
+        val fractionDuration =
+          (delay.max(0.milliseconds) * config.fractionOfDelay).toMillis
+        if (fractionDuration == 0) {
+          delay
+        } else {
+          val randomized = ThreadLocalRandom
+            .current()
+            .nextLong(-fractionDuration, fractionDuration)
+          val randomFactor = FiniteDuration(randomized, TimeUnit.MILLISECONDS)
+          delay + randomFactor
+        }
+      }
+
+      /** Default jitter config which will keep random jitter in +/-10% range
+        */
+      val defaultConfig: RandomJitterConfig = buildJitterConfig(0.2).get
```

Comment:
This looks like +/- 20% because of `nextLong(-fractionDuration, fractionDuration)`